scripts/lib/sync_engine.py

```
@@ -889,29 +889,31 @@ def build_facets(
-            except OSError:
-                pass
-        if pkgs_hash != prev_hash:
-            try:
-                hash_file.parent.mkdir(parents=True, exist_ok=True)
-                hash_file.write_text(pkgs_hash, encoding="utf-8")
-            except OSError:
-                pass
-            import time
+            pass
+    compositions_str = ",".join(sorted(path.name for path in yamls))
+    pkgs_hash = hashlib.md5(f"{pkgs_str}|{compositions_str}".encode()).hexdigest()
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)